plugins/providers/gcs/provider.go

```diff
@@ -0,0 +1,253 @@
+		if utils.ContainsString(resourceTypes, ResourceTypeBucket) {
+			resources = append(resources, bucketResource)
+		}
+	}
+
+	return resources, nil
+}
+
+func (p *Provider) GetType() string {
+	return p.typeName
+}
+
+func (p *Provider) GrantAccess(pc *domain.ProviderConfig, a *domain.Appeal) error {
+	if err := validateProviderConfigAndAppealParams(pc, a); err != nil {
+		return fmt.Errorf("invalid provider/appeal config: %w", err)
+	}
+
+	permissions, err := getPermissions(pc.Resources, a)
+	if err != nil {
+		return fmt.Errorf("error in getting permissions: %w", err)
+	}
+
+	var creds Credentials
+	if err := mapstructure.Decode(pc.Credentials, &creds); err != nil {
+		return fmt.Errorf("error in decoding credentials%w", err)
+	}
+
+	if err := creds.Decrypt(p.crypto); err != nil {
+		return fmt.Errorf("error in decrypting credentials%w", err)
+	}
+
+	client, err := p.getGCSClient(creds)
+	if err != nil {
+		return fmt.Errorf("error in getting new client: %w", err)
+	}
+
+	identity := fmt.Sprintf("%s:%s", a.AccountType, a.AccountID) // identity is AccountType : AccountID, eg: "serviceAccount:test@email.com"
+	if a.Resource.Type == ResourceTypeBucket {
+		b := new(Bucket)
+		if err := b.fromDomain(a.Resource); err != nil {
+			return fmt.Errorf("from Domain func error: %w", err)
+		}
+		for _, p := range permissions {
+			role := iam.RoleName(string(p))
+			if err := client.GrantBucketAccess(context.TODO(), *b, identity, role); err != nil {
+				if errors.Is(err, ErrPermissionAlreadyExists) {
+					return nil
+				}
+				return err
+			}
+		}
+		return nil
+	}
+	return ErrInvalidResourceType
+}
+
+func (p *Provider) RevokeAccess(pc *domain.ProviderConfig, a *domain.Appeal) error {
+	if err := validateProviderConfigAndAppealParams(pc, a); err != nil {
+		return fmt.Errorf("invalid provider/appeal config: %w", err)
+	}
+
+	permissions, err := getPermissions(pc.Resources, a)
+	if err != nil {
+		return fmt.Errorf("error in getting permissions: %w", err)
+	}
+
+	var creds Credentials
+	if err := mapstructure.Decode(pc.Credentials, &creds); err != nil {
+		return fmt.Errorf("error in decoding credentials%w", err)
+	}
+
+	if err := creds.Decrypt(p.crypto); err != nil {
+		return fmt.Errorf("error in decrypting credentials%w", err)
+	}
+
+	client, err := p.getGCSClient(creds)
+	if err != nil {
+		return fmt.Errorf("error in getting new client: %w", err)
+	}
+
+	user := a.AccountID
+	userType := a.AccountType
+	identity := fmt.Sprintf("%s:%s", userType, user)
+	ctx := context.TODO()
+	if a.Resource.Type == ResourceTypeBucket {
+		b := new(Bucket)
+		if err := b.fromDomain(a.Resource); err != nil {
+			return fmt.Errorf("from Domain func error: %w", err)
+		}
+		for _, p := range permissions {
+			var role iam.RoleName = iam.RoleName(string(p))
+			if err := client.RevokeBucketAccess(ctx, *b, identity, role); err != nil {
+				if errors.Is(err, ErrPermissionAlreadyExists) {
+					return nil
+				}
+				return err
+			}
+		}
+		return nil
+	}
```

Comment:
can just pass the context.TODO directly, no need to introduce a new variable if it is only being used in a single place
```suggestion
	if a.Resource.Type == ResourceTypeBucket {
		b := new(Bucket)
		if err := b.fromDomain(a.Resource); err != nil {
			return fmt.Errorf("from Domain func error: %w", err)
		}
		for _, p := range permissions {
			var role iam.RoleName = iam.RoleName(string(p))
			if err := client.RevokeBucketAccess(context.TODO(), *b, identity, role); err != nil {
				if errors.Is(err, ErrPermissionAlreadyExists) {
					return nil
				}
				return err
			}
		}
		return nil
	}
```

Reply:
done in https://github.com/odpf/guardian/pull/227/commits/575cacf66a8181ddf2953708d3c7d26f0b98eba5